src/compiler/utilities.ts

```diff
@@ -5275,9 +5275,10 @@ namespace ts {
         return symbol.flags & SymbolFlags.Transient ? (<TransientSymbol>symbol).checkFlags : 0;
     }
 
-    export function getDeclarationModifierFlagsFromSymbol(s: Symbol): ModifierFlags {
+    export function getDeclarationModifierFlagsFromSymbol(s: Symbol, isWrite = false): ModifierFlags {
         if (s.valueDeclaration) {
-            const flags = getCombinedModifierFlags(s.valueDeclaration);
+            const declaration = (isWrite && forEach(s.declarations, d => d.kind === SyntaxKind.SetAccessor && d)) || s.valueDeclaration;

```

Comment:
```suggestion
            const declaration = (isWrite && find(s.declarations, d => d.kind === SyntaxKind.SetAccessor)) || s.valueDeclaration;
```

Reply:
TIL `forEach`, but not `find`, accepts `undefined` in the first argument

Reply:
Layers of history at work there!